Tests/AppTests/IngestorTests.swift

```diff
@@ -594,4 +597,92 @@ class IngestorTests: AppTestCase {
+                          processingStage: .analysis)
+        let forkedPkg = Package(
+            url: "https://github.com/taz/bar.git",
+            processingStage: .reconciliation
+        )
+        try await pkg.save(on: app.db)
+        try await forkedPkg.save(on: app.db)
+        Current.fetchMetadata = { _, owner, repository in
+                .mock(owner: owner, repository: repository, parentUrl: "https://github.com/foo/bar.git")
+        }
+
+        // MUT
+        try await ingest(client: app.client, database: app.db, mode: .limit(1))
+        
+        guard let forkedId = forkedPkg.id else {
+            XCTFail("Failed to get forked package id")
+            return
+        }
+        
+        guard let id = pkg.id else {
+            XCTFail("Failed to get package id")
+            return
+        }
+        
+        let repo = try await Repository
+            .query(on: app.db)
+            .filter(\Repository.$package.$id == forkedId).first()
+        
+        XCTAssertNotNil(repo?.forkedFrom)
+        
+        XCTAssertEqual(repo?.forkedFrom, .parentId(id))
+        
+    }
+    
+    func test_ingest_storeForkedFromPackageNotInSPI() async throws {
+        let forkedPkg = Package(
+            url: "https://github.com/taz/bar.git",
+            processingStage: .reconciliation
+        )
+        try await forkedPkg.save(on: app.db)
+        Current.fetchMetadata = { _, owner, repository in 
+            .mock(owner: owner, repository: repository, parentUrl: "https://github.com/foo/bar.git")
+        }
+
+        // MUT
+        try await ingest(client: app.client, database: app.db, mode: .limit(1))
+        
+        guard let forkedId = forkedPkg.id else {
+            XCTFail("Failed to get forked package id")
+            return
+        }
+        
+        let repo = try await Repository
+            .query(on: app.db)
+            .filter(\Repository.$package.$id == forkedId).first()
+        
+        XCTAssertNotNil(repo?.forkedFrom)
+        
+        XCTAssertEqual(repo?.forkedFrom, .parentURL("https://github.com/foo/bar.git"))
+    }
+    
+    func test_ingest_storeForkedFromShouldNeNil() async throws {
+        let forkedPkg = Package(
+            url: "https://github.com/taz/bar.git",
+            processingStage: .reconciliation
+        )
+        try await forkedPkg.save(on: app.db)
+        Current.fetchMetadata = { _, owner, repository in
+                .mock(owner: owner, repository: repository, parentUrl: nil)
+        }
+
+        // MUT
+        try await ingest(client: app.client, database: app.db, mode: .limit(1))
+        
+        guard let forkedId = forkedPkg.id else {
+            XCTFail("Failed to get forked package id")
+            return
+        }
+        
+        let repo = try await Repository
+            .query(on: app.db)
+            .filter(\Repository.$package.$id == forkedId).first()
+        
+        XCTAssertNil(repo?.forkedFrom)
+    }
```

Comment:
By moving the main functionality to `getFork`, these tests become much easier to set up because we don't need to test all of `ingest`. Instead we just test `getFork` in isolation:

```swift
    func test_getFork() async throws {
        try await Package(id: .id0, url: "https://github.com/foo/parent.git".url, processingStage: .analysis).save(on: app.db)
        try await Package(url: "https://github.com/bar/forked.git", processingStage: .analysis).save(on: app.db)

        var metadata = Github.Metadata.mock

        do {  // test lookup when package is in the index
            metadata.repository?.parent = .init(url: "https://github.com/foo/parent.git")
            let fork = try await getFork(on: app.db, metadata: metadata)
            XCTAssertEqual(fork, .parentId(.id0))
        }

        do {  // test lookup when package is in the index but with different case in URL
            metadata.repository?.parent = .init(url: "https://github.com/Foo/Parent.git")
            let fork = try await getFork(on: app.db, metadata: metadata)
            XCTAssertEqual(fork, .parentId(.id0))
        }

        do {  // test lookup when package is not in the index
            metadata.repository?.parent = .init(url: "https://github.com/some/other.git")
            let fork = try await getFork(on: app.db, metadata: metadata)
            XCTAssertEqual(fork, .parentURL("https://github.com/some/other.git"))
        }

        do {  // test lookup when parent url is nil
            metadata.repository?.parent = .init(url: nil)
            let fork = try await getFork(on: app.db, metadata: metadata)
            XCTAssertEqual(fork, nil)
        }
    }
```

and then we simply extend a single test where we ensure `fork` is written to the db in `updateRepository`. For example, I'd modify `test_updateRepository_update` and add fork (call on or around line 152):

```swift
        try await updateRepository(on: app.db,
                                   for: repo,
                                   metadata: md,
                                   licenseInfo: .init(htmlUrl: "license url"),
                                   readmeInfo: .init(etag: "etag",
                                                     html: "readme html",
                                                     htmlUrl: "readme html url",
                                                     imagesToCache: []),
                                   s3Readme: .cached(s3ObjectUrl: "url", githubEtag: "etag"),
                                   fork: .parentURL("https://github.com/foo/bar.git"))
```

and validate it (around line 168):

```swift
            XCTAssertEqual(repo.forkedFrom, .parentURL("https://github.com/foo/bar.git"))
```